advntr/vntr_finder.py

```diff
@@ -366,23 +366,52 @@ def check_if_pacbio_read_spans_vntr(self, sema, read, length_distribution, spann
         sema.release()
 
     def check_if_pacbio_mapped_read_spans_vntr(self, sema, read, length_distribution, spanning_reads):
-        flanking_region_size = 100
-        region_start = self.reference_vntr.start_point - flanking_region_size
-        region_end = self.reference_vntr.start_point + self.reference_vntr.get_length()
-        if read.get_reference_positions()[0] < region_start and read.get_reference_positions()[-1] > region_end:
+        hmm_flanking_region_size = 100
+        min_flanking_bp = 10
+        vntr_start = self.reference_vntr.start_point
+        vntr_end = self.reference_vntr.start_point + self.reference_vntr.get_length()
+
+        region_start = vntr_start - hmm_flanking_region_size
+
+        first_aligned_position = read.get_reference_positions()[0]
+        last_aligned_position = read.get_reference_positions()[-1]
+        if first_aligned_position <= vntr_start - min_flanking_bp and vntr_end + min_flanking_bp < last_aligned_position:
             read_region_start = None
             read_region_end = None
-            for read_pos, ref_pos in enumerate(read.get_reference_positions()):
-                if ref_pos >= region_start and read_region_start is None:
-                    read_region_start = read_pos
-                if ref_pos >= region_end and read_region_end is None:
-                    read_region_end = read_pos
+
+            tr_spanning_bp = 0
+            left_flanking_bp = 0
+            right_flanking_bp = 0
+            for read_pos, ref_pos in enumerate(read.get_reference_positions(full_length=True)):
+                if ref_pos is None:  # soft-clip or insertions
+                    continue
+                if ref_pos > vntr_end + hmm_flanking_region_size:  # Boundary check
+                    break
+
+                if region_start <= ref_pos < vntr_end + hmm_flanking_region_size:  # HMM-modeled VNTR region
+                    if region_start <= ref_pos < vntr_start:
+                        if read_region_start is None:
+                            read_region_start = read_pos
+                        left_flanking_bp += 1
+                    elif vntr_start <= ref_pos < vntr_end:
+                        tr_spanning_bp += 1
```

Comment:
Looks like you are not using this after setting it. Is this equal to `read_region_end - read_region_end`? Or can it have a different value because of the None values in the reference positions?

Reply:
They are used later to filter out reads. line 403-406

Reply:
I mean specifically `tr_spanning_bp`